src/frequenz/dispatch/_dispatcher.py

```diff
@@ -286,6 +348,19 @@ def client(self) -> Client:
         """Return the client."""
         return self._client
 
+    @override
+    async def __aenter__(self) -> Self:
+        """Enter an async context.
+
+        Start this background service.
+
+        Returns:
+            This background service.
+        """
+        self.start()
+        await self.wait_for_initialization()
+        return self
```

Comment:
Just in case the parent do something extra, this should be safer:

```suggestion
        await super.__aenter__()
        await self.wait_for_initialization()
        return self
```